### PR TITLE
add displayname validation

### PR DESCRIPTION
**Pull request by netcriptus**
## Uses displayname info to verify the user signature

Fixes: #6954

This PR adds a feature that verifies the user identity using the user id and the displayname received from the PFS.

### CODE REVIEW

File: raiden/tests/unit/transfer/test_source_routing.py [hunk 1]
```diff
@@ -79,6 +82,30 @@ def test_metadata_hashing():
...
+        route=route,
+        address_metadata={signer.address: {"user_id": user_id, "displayname": displayname}},
+    )
+
+    assert bad_metadata.address_metadata == {}
+    assert good_metadata.address_metadata == {
+        signer.address: {"user_id": user_id, "displayname": displayname}
+    }
+
+
```
Comment (fredo) [inline raiden/tests/unit/transfer/test_source_routing.py:108]: Maybe a test where a false verifaication of displayname would remove the metadata would be nice to be covered here

**Review by fredo — CHANGES_REQUESTED** [review]

**Comment by fredo** [comment]
LGTM, one missing path of test coverage would be nice to have

File: raiden/tests/unit/transfer/test_source_routing.py (hunk 1 above)
Comment (netcriptus) [reply raiden/tests/unit/transfer/test_source_routing.py:108]: That's what lines 94 and 103 are about. Or did you have something different in mind?

**Review by fredo — APPROVED** [review]

File: raiden/tests/unit/transfer/test_source_routing.py (hunk 1 above)
Comment (fredo) [reply raiden/tests/unit/transfer/test_source_routing.py:108]: my bad. I dont know what I saw there before...

**Pull request merged**